syne_tune/optimizer/schedulers/multiobjective/multi_surrogate_multi_objective_searcher.py

```diff
@@ -0,0 +1,155 @@
+    decode_state,
+)
+from syne_tune.optimizer.schedulers.searchers.model_based_searcher import (
+    BayesianOptimizationSearcher,
+)
+from syne_tune.optimizer.schedulers.searchers.utils import make_hyperparameter_ranges
+
+logger = logging.getLogger(__name__)
+
+
+class MultiObjectiveMultiSurrogateSearcher(BayesianOptimizationSearcher):
+    """Multi Objective Multi Surrogate Searcher for FIFO scheduler
+
+    This searcher must be used with
+    :class:`~syne_tune.optimizer.schedulers.FIFOScheduler`. It provides
+    Bayesian optimization, based on a scikit-learn estimator based surrogate model.
+
+    Additional arguments on top of parent class
+    :class:`~syne_tune.optimizer.schedulers.searchers.StochasticSearcher`:
+
+    :param estimator: Instance of
+        :class:`~syne_tune.optimizer.schedulers.searchers.bayesopt.sklearn.estimator.SKLearnEstimator`
+        to be used as surrogate model
+    :param scoring_class_and_args: The scoring function (or acquisition
+        function) class and any extra parameters used to instantiate it. If
+        ``None``, expected improvement (EI) is used. Note that the acquisition
+        function is not locally optimized with this searcher.
+    :param num_initial_candidates: Number of candidates sampled for scoring with
+        acquisition function.
+    :param num_initial_random_choices: Number of randomly chosen candidates before
+        surrogate model is used.
+    :param allow_duplicates: If ``True``, allow for the same candidate to be
+        selected more than once.
+    :param restrict_configurations: If given, the searcher only suggests
+        configurations from this list. If ``allow_duplicates == False``,
+        entries are popped off this list once suggested.
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: List[str],
+        estimators: Dict[str, Estimator],
+        mode: List[str] = None,
+        points_to_evaluate: Optional[List[Dict[str, Any]]] = None,
+        scoring_class_and_args: Optional[ScoringClassAndArgs] = None,
+        num_initial_candidates: int = DEFAULT_NUM_INITIAL_CANDIDATES,
+        num_initial_random_choices: int = DEFAULT_NUM_INITIAL_RANDOM_EVALUATIONS,
+        allow_duplicates: bool = False,
+        restrict_configurations: Optional[List[Dict[str, Any]]] = None,
+        clone_from_state: bool = False,
+        **kwargs,
+    ):
+        if mode is None:
+            mode = ["min" for item in metric]
+
+        super().__init__(
+            config_space,
+            metric=metric,
+            mode=mode,
+            points_to_evaluate=points_to_evaluate,
+            random_seed_generator=kwargs.get("random_seed_generator"),
+            random_seed=kwargs.get("random_seed"),
+        )
+        self.estimator = estimators
+        if scoring_class_and_args is None:
+            scoring_class_and_args = MultiObjectiveLCBRandomLinearScalarization
```

Comment:
Please use this, so that the random sampling in the acq func is controlled:

Higher up:
`from syne_tune.optimizer.schedulers.searchers.searcher_base import extract_random_seed`

Below:

```
if scoring_class_and_args is None:
    random_seed = extract_random_seed(**kwargs)
    scoring_class_and_args = (MultiObjectiveLCBRandomLinearScalarization, {"random_seed": random_seed})
```

Reply:
Together with my comment below, this will work